## Taking input from user

In [1]:
#using input() function

input()

10


'10'

In [2]:
input("Enter your name")

Enter your nameshailesh


'shailesh'

In [11]:
myname=input("Enter your name")
print("My name is {}".format(myname))

Enter your nameshail
My name is shail


### take two numbers from user and find the sum

In [13]:
first_num=input("Enter first number")
second_num=input("Enter second number")

result=first_num+second_num
print("The sum of two numbers is\t",result)

Enter first number10
Enter second number20
The sum of two numbers is	 1020


#### conclusion: The sum of 10 and 20 should have been 30 however we are getting 1020. By default the input from user is in string therefore summation means concatenation of two strings not the addition of two integers

### To solve above problem, we need to understand the concept of type conversion

## TYPE CONVERSION

### Process of converting a data type into another data type- unless it is not convertible

##### There are two types of type conversion- implicit and explicit

In [22]:
#implicit
3.5+5

8.5

##### Here implicitly the result is in decimal format- implicilty sum of a integer and a float is converted into  float

In [23]:
5+ 6+2j

(11+2j)

In [24]:
4.5 + 4+5j

(8.5+5j)

In [26]:
#Explicit type conversion
# Here 4.5 will be converted into 4
int(4.5)

4

In [27]:
float(4)

4.0

In [28]:
str(4)

'4'

In [29]:
bool(1)

True

In [30]:
bool(4)

True

In [31]:
bool(0)

False

In [32]:
complex(4)

(4+0j)

In [33]:
complex(4.5)

(4.5+0j)

In [35]:
list('hello')

['h', 'e', 'l', 'l', 'o']

In [37]:
int('kolkata')

ValueError: invalid literal for int() with base 10: 'kolkata'

###### Type conversion is not a permannent operation. On the other hand type casting is a permanent operation

In [38]:
first_num=input("Enter first number")
second_num=input("Enter second number")

result=int(first_num)+int(second_num)
print("The sum of two numbers is\t",result)

Enter first number10
Enter second number20
The sum of two numbers is	 30


#### We solved the summation issue using concept of type conversion